.teamcity/hosted/arm/PushStagingLinux2004_Aarch64.kts

```diff
@@ -0,0 +1,78 @@
+package generated
+
+import common.TeamCityDockerImagesRepo
+import hosted.utils.ImageInfoRepository
+import hosted.utils.steps.buildAndPublishImage
+import jetbrains.buildServer.configs.kotlin.v2019_2.AbsoluteId
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import jetbrains.buildServer.configs.kotlin.v2019_2.FailureAction
+import jetbrains.buildServer.configs.kotlin.v2019_2.ReuseBuilds
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.dockerSupport
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.freeDiskSpace
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.swabra
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.dockerCommand
+
+/**
+ * Building and deploying aarch64 (ARM) Linux-based Docker images into staging registry, which is defined ...
+ * ... within upstream's parameters.
+ */
+object push_staging_linux_2004_aarch64 : BuildType({
+    name = "[aarch64] [Linux 2004] Build and deploy staging images"
+    buildNumberPattern = "%dockerImage.teamcity.buildNumber%-%build.counter%"
+    description = "Build and deploy Linux-based Docker images for aarch64 platform."
+    vcs {
+        root(TeamCityDockerImagesRepo.TeamCityDockerImagesRepo)
+    }
+
+    steps {
+        dockerCommand {
+            name = "pull ubuntu:20.04"
+            commandType = other {
+                subCommand = "pull"
+                commandArgs = "ubuntu:20.04"
+            }
+        }
+
+        ImageInfoRepository.getArmImages().forEach { imageInfo ->
+            buildAndPublishImage(imageInfo)
+        }
+    }
+
+    features {
+        freeDiskSpace {
+            requiredSpace = "8gb"
+            failBuild = true
+        }
+
+        dockerSupport {
+            cleanupPushedImages = true
```

Comment:
AFAIK, previously we haven't cleaned images for the staging repository. Could you please re-check?

Reply:
Looking at other DSLs and their generator, cleanup of the images seems to be enabled for staging registry:
1. [TeamCityKotlinSettingsGenerator.cs](https://github.com/JetBrains/teamcity-docker-images/blob/development/2023.05.x/tool/TeamCity.Docker/TeamCityKotlinSettingsGenerator.cs)
```
        private IEnumerable<string> CreateDockerFeature()
        {
            …
            yield return "dockerSupport {";
            yield return "\t cleanupPushedImages = true";
            ….
       }
```
2. [BuildAndPushHosted](https://github.com/JetBrains/teamcity-docker-images/blob/development/2023.05.x/.teamcity/hosted/BuildAndPushHosted.kts):
```
 dockerSupport {
            cleanupPushedImages = true
            …
            }
```
Could you, please, suggest if that'd useful for us to disable the cleanup? 
Personally, I would keep the cleanup of staging images, as, based on my understanding, we don't have use-cases when staging images are intended to be kept within the registry for a relatively long period of time (perhaps I'm missing something?).